deepsearcher/online_query.py

```diff
@@ -9,33 +9,41 @@
 import asyncio
 
 
-
 # Add a wrapper function to support synchronous calls
 def query(original_query: str, max_iter: int = 3) -> Tuple[str, List[RetrievalResult]]:
-    return asyncio.run(async_query(original_query, max_iter))
+    a, b, _ = asyncio.run(async_query(original_query, max_iter))
+    return a, b
 
+def query_with_consumed_token(original_query: str, max_iter: int = 3) -> Tuple[str, List[RetrievalResult], int]:
+    return asyncio.run(async_query(original_query, max_iter))
 
-async def async_query(original_query: str, max_iter: int = 3) -> Tuple[str, List[RetrievalResult]]:
-    retrieval_res, all_sub_queries = await async_retrieve(original_query, max_iter)
+async def async_query(original_query: str, max_iter: int = 3) -> Tuple[str, List[RetrievalResult], int]:
+    retrieval_res, all_sub_queries, retrieve_conseumed_token = await async_retrieve(original_query, max_iter)
     ### GENERATE FINAL ANSWER ###
     log.color_print("<think> Generating final answer... </think>\n")
-    final_answer = generate_final_answer(original_query, all_sub_queries, retrieval_res)
+    final_answer, final_consumed_token = generate_final_answer(original_query, all_sub_queries, retrieval_res)
     log.color_print("\n==== FINAL ANSWER====\n")
     log.color_print(final_answer)
-    return final_answer, retrieval_res
+    return final_answer, retrieval_res, retrieve_conseumed_token + final_consumed_token
+
 
+def retrieve(original_query: str, max_iter: int = 3) -> Tuple[str, List[RetrievalResult], int]:
+    a, b, _ = asyncio.run(async_retrieve(original_query, max_iter))
+    return a, b
 
-def retrieve(original_query: str, max_iter: int = 3) -> Tuple[str, List[RetrievalResult]]:
+def retrieve_with_consumed_token(original_query: str, max_iter: int = 3) -> Tuple[str, List[RetrievalResult], int]:
```

Comment:
maybe this method is redudant. Since the return type of `async_retrieve` method changed from `Tuple[List[RetrievalResult], List[str]` to `Tuple[List[RetrievalResult], List[str], int]`, we can directly change the return type of `retrieve()` in the same way, without adding `retrieve_with_consumed_token()` method